src/main/java/io/cryostat/net/web/http/api/v2/DiscoveryPostHandler.java

```diff
@@ -131,6 +132,8 @@ public IntermediateResponse<Void> handle(RequestParameters params) throws Except
             return new IntermediateResponse<Void>().body(null);
         } catch (JsonSyntaxException | IllegalArgumentException e) {
```

Comment:
In the case of schema violation (e.g. null primary key), I am a bit unsure which Exception will be thrown. Same for `DiscoveryRegistrationHandler`?

Reply:
Should we also add these cases of violation to `DiscoveryPluginIT` test?

Reply:
The `PluginInfo` primary key is a UUID generated by the database, so it can't be nulled by the registering client. If they try to register with a null/empty realm or callback URI then it's an` IllegalArgumentException` via `requireNonBlank`. If the subtree they provide is not a valid JSON document then it's a `JsonSyntaxException`.